include/seqan3/range/views/pseudo_random_access.hpp

```diff
@@ -0,0 +1,199 @@
+
+        if constexpr (std::ranges::random_access_range<urng_t>)
+        { // Nothing to do, just return as ref_view or original view.
+            return std::views::all(std::forward<urng_t>(urange));
+        }
+        else
+        { // Get a subrange using the random access iterators of the container.
+            using iterator_ra_t = decltype(std::declval<urng_t>().begin_ra());
+            using sentinel_ra_t = decltype(std::declval<urng_t>().end_ra());
+
+            return std::ranges::subrange<iterator_ra_t, sentinel_ra_t>{urange.begin_ra(), urange.end_ra()};
+        }
+    }
+};
+
+}  // namespace seqan3::detail
+
+namespace seqan3::views
+{
+
+/*!\name General purpose views
+ * \{
+ */
+
+/*!\brief            A view adaptor that converts a pseudo random access range to a std::random_access_range.
+ * \tparam urng_t    The type of the range being processed. See below for requirements. [template parameter is
+ *                   omitted in pipe notation]
+ * \param[in] urange The range being processed. [parameter is omitted in pipe notation]
+ * \returns          A std::ranges::random_access_range over the given range.
+ * \ingroup views
+ *
+ * \details
+ *
+ * **Header**
+ * ```cpp
+ *      #include <seqan3/range/views/pseudo_random_access.hpp>
+ * ```
+ *
+ * A pseudo random access range is a range, whose iterator typically defines all the interfaces necessary to allow
+ * random access, but cannot guarantee accessing an arbitrary element in constant time.
+ * Thus, the highest category it can support by default is std::ranges::bidirectional_range. However, for many of these
+ * pseudo random access ranges better algorithms and data structures with sub-linear runtime complexities can be used
+ * (for example logarithmic time complexity). To enforce the faster behaviour of the range in a generic
+ * range-based context you can use this range adaptor, which will return a range that models
+ * std::ranges::random_access_range. Note, that this does not mean that the complexity of accessing an arbitrary element
+ * of the adapted range improves to constant time, but merely all syntactical requirements are fulfilled including the
+ * iterator tag.
+ *
+ * ### View properties
+ *
+ * | range concepts and reference_t   | `urng_t` (underlying range type)  | `rrng_t` (returned range type)             |
+ * |----------------------------------|:---------------------------------:|:------------------------------------------:|
+ * | std::ranges::input_range         | *required*                        | *preserved*                                |
+ * | std::ranges::forward_range       |                                   | *preserved*                                |
+ * | std::ranges::bidirectional_range |                                   | *preserved*                                |
+ * | std::ranges::random_access_range |                                   | *preserved*                                |
```

Comment:
```suggestion
 * | std::ranges::random_access_range |                                   | *guaranteed*                              |
```

And also the weaker concepts